### PR TITLE
SUP-32497 download isOriginal instead of parmsId 0 for source flavora…

### PR DESCRIPTION
**Pull request by bw-kaltura**
…sset - use case static conversion profile

### CODE REVIEW

File: alpha/lib/model/entry.php [hunk 1]
```diff
@@ -3117,15 +3117,27 @@ public function createDownloadAsset(BatchJob $parentJob = null, $flavorParamsId,
...
-		$flavorAsset = assetPeer::retrieveByEntryIdAndParams($this->getId(), $flavorParamsId);
-		if ($flavorAsset && $flavorAsset->getStatus() == flavorAsset::FLAVOR_ASSET_STATUS_READY)
-			return $flavorAsset->getDownloadUrl();
-			
-		if(assetPeer::retrieveOriginalByEntryId($this->getId()))
-			return null; // flavor asset should be created
-			
-		// entry file sync should be used (data or download)
-		return $this->getRawDownloadUrl();
+        	if($flavorParamsId == 0)
```
Comment (inbal-ben-david) [inline alpha/lib/model/entry.php:3120]: ===

File: alpha/lib/model/entry.php [hunk 2]
```diff
@@ -3117,15 +3117,27 @@ public function createDownloadAsset(BatchJob $parentJob = null, $flavorParamsId,
...
+        	{
+        	    $flavorAsset = assetPeer::retrieveByEntryIdAndParams($this->getId(), $flavorParamsId);
+        	}
+	
+        	if($flavorAsset && $flavorAsset->getStatus() == flavorAsset::ASSET_STATUS_READY)
+        	{
+        	    return $flavorAsset->getDownloadUrl();
+        	}
+	
+        	if($flavorParamsId != 0 && assetPeer::retrieveOriginalByEntryId($this->getId()))
```
Comment (inbal-ben-david) [inline alpha/lib/model/entry.php:3134]: remove $flavorParamsId != 0

File: alpha/lib/model/entry.php (hunk 1 above)
Comment (bw-kaltura) [reply alpha/lib/model/entry.php]: done

File: alpha/lib/model/entry.php (hunk 2 above)
Comment (bw-kaltura) [reply alpha/lib/model/entry.php]: done

**Review by inbal-ben-david — APPROVED** [review]

File: alpha/lib/model/entry.php (hunk 1 above)
Comment (bw-kaltura) [reply alpha/lib/model/entry.php]: Changed to === "0"
$flavorParamsId is string.

Comment (jessp01) [reply alpha/lib/model/entry.php]: Hi @bw-kaltura ,

I'm a bit confused as to why it is a string? Doxygen block declares it to be an `int`:
https://github.com/kaltura/server/blob/f04763dd877c7df2f2dd396b08d9e985c4a55dcf/alpha/lib/model/entry.php#L3115

The `flavor_asset.flavor_params_id` column is also defined as `int`:
https://github.com/kaltura/server/blob/Rigel-18.9.0/deployment/base/sql/01.kaltura_ce_tables.sql#L1073

And lastly:
https://github.com/kaltura/server/blob/Rigel-18.9.0/alpha/lib/model/om/BaseflavorParamsConversionProfile.php#L172

**Review by inbal-ben-david — APPROVED** [review]

**Pull request merged**